packages/hoppscotch-app/src/helpers/actions.ts

```diff
@@ -55,8 +57,8 @@ export function bindAction(action: HoppAction, handler: () => void) {
   activeActions$.next(Object.keys(boundActions) as HoppAction[])
 }
 
-export function invokeAction(action: HoppAction) {
-  boundActions[action]?.forEach((handler) => handler())
+export function invokeAction(action: HoppAction, args?: any[]) {
```

Comment:
The action system is not supposed to be having an `args` parameter, even if allowed which can be arranged, I am not a fan of it being there without type safety.

Reply:
@AndrewBastin could you provide further details about the action arguments solution you mentioned in your first comment? I think that usingarguments in action handlers allow them to be more flexible and prevent scenarios of having to store data in a more global scope.

Reply:
Yeah, well, the thing is we don't want to use `any` in our codebase as much as possible as it removes all the contract guarantees TypeScript can enforce and this leads to our code being more prone to errors. Hence, we try to type check all our code as much as possible to make sure our code is free of errors. For example, in the review only I pointed out how accidentally calling with the `envName` not being "Global" or something is totally possible.

No worries, I have some time today to come up with a type that can express the action handler types safely.

Reply:
I hae pushed the update which implements this.